cloudflare/resource_cloudflare_filter.go

```diff
@@ -111,6 +111,26 @@ func resourceCloudflareFilterCreate(d *schema.ResourceData, meta interface{}) er
 func resourceCloudflareFilterRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*cloudflare.API)
 	zoneID := d.Get("zone_id").(string)
+	zoneName := d.Get("zone").(string)
+
+	if zoneID == "" {
+		zoneID, _ = client.ZoneIDByName(zoneName)
+	} else {
+		zones, err := client.ListZones()
+		if err != nil {
+			return fmt.Errorf("failed to lookup all zones: %s", err)
+		}
+
+		for _, zone := range zones {
+			if zone.ID == zoneID {
+				zoneName = zone.Name
+			}
+		}
+
+		if zoneName == "" {
```

Comment:
this is a safe guard to ensure that if we do get to this point, we really do have a `zoneName` set

Reply:
We should not need to check for `zoneID` being empty here, will be always populated by `Create` or `Import`.
We should also mark `zone` field as `Computed: true`.

Reply:
`Computed` here means that the code may set it itself, so it should not be taken into consideration on diff unless explicitly set (yes, `Computed` fields can be set in HCL).

Because now in `Create` we set `zone_id` if we provide `zone`, but we will be setting `zone` when we import with `zone_id` only, we need both to be `Computed`.

I made a mistake coding in the `zone` field, I was young and stupid at that time ;)
We really need to clean `zone` vs `zone_id` mess in v2.0 😰

Reply:
> `Computed` here means that the code may set it itself, so it should not be taken into consideration on diff unless explicitly set (yes, `Computed` fields can be set in HCL).

TIL, other uses of this that I have seen have all been for things like UUIDs or generating values at runtime.

> We really need to clean `zone` vs `zone_id` mess in v2.0 😰

Do you want to start a v2.0 issue where we can keep track of these things so that we have a record of it?

Reply:
Yup, good idea. Issue and/or milestone (will need to read GitHub guides on this).